src/main/java/frc/robot/commandgroups/bits/TestBallFlow.java

```diff
@@ -0,0 +1,52 @@
+package frc.robot.commandgroups.bits;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.commandgroups.PickUpCargo;
+import frc.robot.commandgroups.ShootCargo;
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.shooter.Shooter;
+
+import java.util.function.BooleanSupplier;
+
+public class TestBallFlow extends SequentialCommandGroup {
+    private final Intake intake;
+    private final Shooter shooter;
+    private final Conveyor conveyor;
+    private final ShootCargo shootCargo;
+    private final PickUpCargo pickUpCargo;
+
+    public TestBallFlow(Conveyor conveyor, Intake intake, Shooter shooter,
+                        PickUpCargo pickUpCargo, BooleanSupplier pickUpCargoIsFinished,
+                        ShootCargo shootCargo, BooleanSupplier shootCargoIsFinished) {
```

Comment:
You don't need to pass commands, just their parameters, and then create them internally